src/Symfony/Component/Security/Core/Authorization/SwitchUserAuthorizationChecker.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Core\Authorization;
+
+use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\Role\SwitchUserRole;
+
+/**
+ * SwitchUserAuthorizationChecker.
+ *
+ * It gives access to the token representing the impersonated user.
+ *
+ * @author Fabien Papet <fabien.papet@gmail.com>
+ */
+class SwitchUserAuthorizationChecker implements SwitchUserAuthorizationCheckerInterface
+{
+    /**
+     * @var AccessDecisionManagerInterface
+     */
+    private $accessDecisionManager;
+    /**
+     * @var TokenStorageInterface
+     */
+    private $tokenStorage;
+
+    /**
+     * @param AccessDecisionManagerInterface $accessDecisionManager
+     * @param TokenStorageInterface          $tokenStorage
```

Comment:
imo this PHPDoc is not needed, as the arguments are typehinted, not sure about the member vars PHPDoc

Reply:
removed